clients/src/test/java/org/apache/kafka/common/requests/RequestResponseTest.java

```diff
@@ -843,6 +845,17 @@ public void testApiVersionResponseWithNotUnsupportedError() {
         }
     }
 
+    @Test
+    public void UnregisterBrokerResponseWithNotUnsupportedError() {
+        UnregisterBrokerRequest request = new UnregisterBrokerRequest.Builder(new UnregisterBrokerRequestData())
+                .build((short) 0);
+        UnregisterBrokerResponse response = request.getErrorResponse(0, Errors.INVALID_REQUEST.exception());
```

Comment:
Could you please create a exception with custom error message? otherwise, this test does not ensure the custom exception is added into the response.